nano/secure/ledger.cpp

```diff
@@ -1156,68 +1141,32 @@ nano::account nano::ledger::account (nano::block const & block) const
 	return result;
 }
 
-nano::account nano::ledger::account (store::transaction const & transaction, nano::block_hash const & hash) const
+std::optional<nano::account> nano::ledger::account (store::transaction const & transaction, nano::block_hash const & hash) const
 {
 	auto block_l = block (transaction, hash);
-	debug_assert (block_l != nullptr);
-	return account (*block_l);
-}
-
-nano::account nano::ledger::account_safe (store::transaction const & transaction_a, nano::block_hash const & hash_a, bool & error_a) const
-{
-	if (!pruning)
+	if (!block_l)
 	{
-		return account (transaction_a, hash_a);
-	}
-	else
-	{
-		auto block_l = block (transaction_a, hash_a);
-		if (block_l != nullptr)
-		{
-			return account (*block_l);
-		}
-		else
-		{
-			error_a = true;
-			return 0;
-		}
-	}
-}
-
-nano::account nano::ledger::account_safe (store::transaction const & transaction, nano::block_hash const & hash) const
-{
-	auto block_l = block (transaction, hash);
-	if (block_l)
-	{
-		return account (*block_l);
-	}
-	else
-	{
-		return { 0 };
+		return std::nullopt;
 	}
+	return account (*block_l);
 }
 
 std::optional<nano::account_info> nano::ledger::account_info (store::transaction const & transaction, nano::account const & account) const
 {
 	return store.account.get (transaction, account);
 }
 
-nano::uint128_t nano::ledger::amount (store::transaction const & transaction_a, nano::block_hash const & hash_a)
-{
-	auto block_l = block (transaction_a, hash_a);
-	auto block_balance (balance (transaction_a, hash_a));
-	auto previous_balance (balance (transaction_a, block_l->previous ()));
-	return block_balance > previous_balance ? block_balance - previous_balance : previous_balance - block_balance;
-}
-
-nano::uint128_t nano::ledger::amount_safe (store::transaction const & transaction_a, nano::block_hash const & hash_a, bool & error_a) const
+std::optional<nano::uint128_t> nano::ledger::amount (store::transaction const & transaction_a, nano::block_hash const & hash_a)
 {
 	auto block_l = block (transaction_a, hash_a);
 	debug_assert (block_l);
```

Comment:
Shouldn't we check if block is empty and return nullopt instead of keeping this debug assert?